src/Alerts/atom/ConfirmationModal.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import styled from 'styled-components'
+import Button from '../../Form/atoms/Button';
+import { TypeButtonDesigns } from '../../Form';
+import { useModal } from '../../hooks';
+
+const Container = styled.div``;
+
+const Title = styled.div`
+  text-shadow: 0 0 10px hsla(0, 0%, 100%, 0.8);
+  font-size: 20px;
+  font-weight: 500;
+  font-stretch: normal;
+  font-style: normal;
+  line-height: normal;
+  letter-spacing: normal;
+  color: hsl(210, 6%, 47%);
+`;
+
+const MessageBox = styled.div`
+  font-size: 14px;
+  font-weight: 500;
+  font-stretch: normal;
+  font-style: normal;
+  line-height: 1.79;
+  letter-spacing: normal;
+  color: hsl(200, 1%, 49%);
+  margin: 28px 0;
+`;
+
+const StyledButton = styled(Button)`
+  margin-left: 10px;
+`;
+
+const ButtonsGroup = styled.div`
+  text-align: right;
+`;
+
+
+type IConfirmationModal = {
+  title?: string,
+  message: string,
+  leftButtonText?: string,
+  leftButtonDesign?: TypeButtonDesigns,
+  leftButtonCallback?: () => void,
+  rightButtonText?: string,
+  rightButtonDesign?: TypeButtonDesigns,
+  rightButtonCallback?: () => void,
+}
+
+const ConfirmationModal: React.FC<IConfirmationModal> = ({
+  title = '',
+  message = '',
+  leftButtonText = 'submit',
+  leftButtonDesign = 'primary',
+  leftButtonCallback = () => { },
+  rightButtonDesign = 'secondary',
+  rightButtonText = 'cancel',
+  rightButtonCallback = () => { },
+}) => {
+
+  const { setIsOpen } = useModal();
+
+  const handlePrimaryBtn = () => {
+    leftButtonCallback();
+    setIsOpen(false);
+  }
```

Comment:
With this, is there an easy way to `setIsOpen(false)` from executing if the primary action has failed?

For example, the user wants to delete a file and so we add a checkbox that says, "I confirm I want to delete this file". If they haven't checked it and the user presses the button, we need to be able to keep the modal open.

If this is more complex we can move this to its own improvement issue. Perhaps we need to consider being able to set  enabled / disabled states too. 🤔

Reply:
In that specific scenario I imagine that User should evaluate if the "Form" or "Elements in component" required are valid setting it too close. Something similar to this:

<Button
onClick ={ 
   () => {
      const areFieldsValid = validateFields()
      if(areFieldsValid) { 
            do something ... then setModalOpen(false) 
      }else {
         showInvalidFields();
        }
     }
}> Delete File </Button>

I think allowing the developer to use setModalOpen value gives a lot of power to have this flexible 🤔

Reply:
Is there an example or can we get an example for this? I think this is something we want people to be able to understand as I'm sure it will come up quickly.

Reply:
I have updated